qa/tasks/cephfs/test_nfs.py

```diff
@@ -70,45 +107,173 @@ def _test_delete_cluster(self):
             wait_time += 10
         self.fail("NFS Ganesha cluster could not be deleted")
 
+    def _test_list_cluster(self, empty=False):
+        '''
+        Tests listing of deployed nfs clusters. If nfs cluster is deployed then
+        it checks for expected cluster id. Otherwise checks nothing is listed.
+        :param empty: If true it denotes no cluster is deployed.
+        '''
+        if empty:
+            cluster_id = ''
+        else:
+            cluster_id = self.cluster_id
+        nfs_output = self._nfs_cmd('cluster', 'ls')
+        self.assertEqual(cluster_id, nfs_output.strip())
+
     def _create_export(self, export_id, create_fs=False, extra_cmd=None):
+        '''
+        Tests creation of a single export.
+        :param export_id: Denotes export number
+        :param create_fs: If false filesytem exists. Otherwise create it.
+        :param extra_cmd: List of extra arguments for creating export.
+        '''
         if create_fs:
             self._cmd('fs', 'volume', 'create', self.fs_name)
         export_cmd = ['nfs', 'export', 'create', 'cephfs', self.fs_name, self.cluster_id]
         if isinstance(extra_cmd, list):
             export_cmd.extend(extra_cmd)
         else:
             export_cmd.append(self.pseudo_path)
-
+        # Runs the nfs export create command
         self._cmd(*export_cmd)
-        res = self._sys_cmd(['rados', '-p', 'nfs-ganesha', '-N', self.cluster_id, 'get', f'export-{export_id}', '-'])
+        # Check if user id for export is created
+        self._check_auth_ls(export_id, check_in=True)
+        res = self._sys_cmd(['rados', '-p', 'nfs-ganesha', '-N', self.cluster_id, 'get',
+                             f'export-{export_id}', '-'])
+        # Check if export object is created
         if res == b'':
-            raise RuntimeError("Export cannot be created")
+            self.fail("Export cannot be created")
 
     def _create_default_export(self):
-            self._test_create_cluster()
-            self._create_export(export_id='1', create_fs=True)
+        '''
+        Deploys a single nfs cluster and creates export with default options.
+        '''
+        self._test_create_cluster()
+        self._create_export(export_id='1', create_fs=True)
 
     def _delete_export(self):
+        '''
+        Deletes an export.
+        '''
         self._nfs_cmd('export', 'delete', self.cluster_id, self.pseudo_path)
+        self._check_auth_ls()
+
+    def _test_list_export(self):
+        '''
+        Tests listing of created exports.
+        '''
+        nfs_output = json.loads(self._nfs_cmd('export', 'ls', self.cluster_id))
+        self.assertIn(self.pseudo_path, nfs_output)
+
+    def _test_list_detailed_and_get_export(self):
```

Comment:
Separate into tests for `export ls --detailed` and `export get` commands. 

And test whether `export ls --detailed` works with multiple exports.

Reply:
Done.